app/components/have-a-question.js

```diff
@@ -47,46 +49,77 @@ export default function HaveAQuestion(props) {
       if (response && response.error) {
         let { error } = response
         setResponse(error)
+        setErrorMessage(true)
       } else {
         setResponse('Your question was sucessfully submitted!')
       }
+      setIsAlertVisible(true)
+
+      setTimeout(() => {
+        setIsAlertVisible(false)
```

Comment:
When the operation is successful it works great.  But when "somethings worng" (you don't type in success@email.com) you are suck not being able to try again.

There can be a different setTimeout for the case when there is an error vs when there isn't.

When there is an error, after the 10 seconds, I think we should leave the email address and submit button visible but clear the error message, and allow the user to change the email address if necessary and hit submit again.

Managing the states, and making it easy to understand is a trick thing, but a good thing to get good at. Lets try this a little more.

Reply:
Sure!